state/backups/backups.go

```diff
@@ -206,67 +200,22 @@ func (b *backups) Create(meta *Metadata, paths *Paths, dbInfo *DBInfo, keepCopy,
 		return "", errors.Annotate(err, "while updating metadata")
 	}
 
-	// Store the archive if asked by user
-	if keepCopy {
-		err = storeArchive(b.storage, meta, result.archiveFile)
-		if err != nil {
-			return "", errors.Annotate(err, "while storing backup archive")
-		}
-	}
-
 	return result.filename, nil
 }
 
-type diskUsage struct{}
-
-func (diskUsage) Available(path string) uint64 {
-	return du.NewDiskUsage(path).Available()
-}
-
-// Add stores the backup archive and returns its new ID.
-func (b *backups) Add(archive io.Reader, meta *Metadata) (string, error) {
-	// Store the archive.
-	err := storeArchive(b.storage, meta, archive)
-	if err != nil {
-		return "", errors.Annotate(err, "while storing backup archive")
-	}
-
-	return meta.ID(), nil
-}
-
-// Get retrieves the associated metadata and archive file from model storage.
-// There are two cases, the archive file can be in the juju database or
-// a file on the machine.
-func (b *backups) Get(id string) (*Metadata, io.ReadCloser, error) {
-	if strings.Contains(id, TempFilename) {
-		return b.getArchiveFromFilename(id)
-	}
-	rawmeta, archiveFile, err := b.storage.Get(id)
-	if err != nil {
-		return nil, nil, errors.Trace(err)
-	}
-
-	meta, ok := rawmeta.(*Metadata)
-	if !ok {
-		return nil, nil, errors.New("did not get a backups.Metadata value from storage")
-	}
-
-	return meta, archiveFile, nil
-}
-
-func (b *backups) getArchiveFromFilename(name string) (_ *Metadata, _ io.ReadCloser, err error) {
-	dir, _ := path.Split(name)
-	build := builder{rootDir: dir}
+// Get retrieves the associated metadata and archive file a file on the machine.
+func (b *backups) Get(fileName string) (_ *Metadata, _ io.ReadCloser, err error) {
 	defer func() {
-		if err2 := build.removeRootDir(); err2 != nil {
-			logger.Errorf(err2.Error())
-			if err != nil {
-				err = errors.Annotatef(err2, "getArchiveFromFilename(%s)", name)
-			}
+		// On success, remove the retrieved file.
+		if err != nil {
+			return
+		}
+		if err2 := os.Remove(fileName); err2 != nil && !os.IsNotExist(err2) {
```

Comment:
why err2 here? scope changes.

Reply:
yeah, the original code needed a separate err var but the refactored code could do away with it